proposals/apple-mdm/base-setup-clix.md

```diff
@@ -0,0 +1,205 @@
+
+WARNING: If you change the SCEP CA in the fleet server, you must un-enroll and re-enroll all devices.
+
+Continue? (Y/n)
+
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleet-scep.crt
+Updating ENV["NEED_VARNAME"] with contents of /path/to/fleet-scep.key
+```
+
+## Misc
+### Error States
+#### Missing required flag
+Matches current behavior of `fleetctl`: Simply shows the help page (try `fleetctl setup --name="hello"`)
+#### Required file does not exist 
+Matches current behavior of `fleetctl`: `Error: open does/not/exist: no such file or directory`
+(Try `fleetctl apply -f does/not/exist`)
+### Help pages - Setup
+
+```bash
+$ fleetctl setup
+NAME:
+   fleetctl setup - Set up a Fleet instance or MDM // UPDATED
+
+USAGE:
+   fleetctl setup [options]
+   fleetctl setup apns // NEW
+   fleetctl setup scep // NEW
+   
+SUBCOMMANDS:
+	apns Apple Push Notification Service to enable MDM for Apple devices
+	scep Simple Certificate Enrollment Protocol to enable MDM
+
+OPTIONS:
+	... // remains same
+```
+
+### Help pages - APNS
+
+``` bash
+// ALL NEW
+$ fleetctl setup apns
+NAME:
+	fleetctl setup apns - Set up Apple Push Notification Service to enable Apple MDM
+
+USAGE:
+	fleetctl setup apns command
+
+COMMANDS:
+	init     Generate the necessary files to set up APNS
+	gen-req  Generate the .req file for Apple from the .p7 file from Fleet
+	finalize Set the SCEP certs in the Fleet Server's environment variables
+```
+
+``` bash
+// ALL NEW
+$ fleetctl setup apns init
+NAME:
+	fleetctl setup apns init - Generates the necessary files to set up APNS 
+
+USAGE:
+	fleetctl setup apns init
+```
+
+``` bash
+// ALL NEW
+$ fleetctl setup apns gen-req
+NAME:
+	fleetctl setup apns gen-req - Generate a .req file from a .p7 file provided by FleetDM, Inc.
+
+USAGE:
+	fleetctl setup apns gen-req [options]
+
+OPTIONS:
+  --p7-file value  Path to the .p7 file provided by FleetDM, Inc. (required)
+  --pki-key value  Path to the fleet-apns-pki.key file generated by the `fleetctl setup apns init` command (required)
+  --pki-crt value  Path to the fleet-apns-pki.crt file generated by the `fleetctl setup apns init` command (required)
+  --context value  Name of fleetctl config context to use (default: "default") [$CONTEXT]
+```
+
+``` bash
+// ALL NEW
+$ fleetctl setup apns finalize
+NAME:
+	fleetctl setup apns finalize - Set the APNS certs in the Fleet Server's environment variables
+
+USAGE:
+	fleetctl setup apns finalize [options]
+
+OPTIONS:
+	--apple-crt value Path to the .crt file provided by Apple (required)
+	--push-key value  Path to the fleet-apns-push.key file generated by the `fleetctl setup apns init` command (required)
+  --context value   Name of fleetctl config context to use (default: "default") [$CONTEXT]
+```
+
+### Help Pages - SCEP
+
+``` bash
+// ALL NEW
+$ fleetctl setup scep
+NAME:
+	fleetctl setup scep - Set up Simple Certificate Enrollment Protocol to enable MDM
```

Comment:
```suggestion
	fleetctl setup scep - Set up Simple Certificate Enrollment Protocol to enable MDM for Apple devices
```
Same comment:

Reply:
I guess we should do this anyway, since we are only doing apple devices. Could remove this later.